src/daft-local-execution/src/lib.rs

```diff
@@ -45,12 +89,34 @@ impl<T: 'static> TaskSet<T> {
     }
 }
 
-pub struct ExecutionRuntimeHandle {
+struct SpawnedTask<T>(tokio::task::JoinHandle<T>);
+impl<T> Future for SpawnedTask<T> {
+    type Output = crate::Result<T>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
+        let this = self.get_mut();
+        Pin::new(&mut this.0).poll(cx).map(|r| r.context(JoinSnafu))
```

Comment:
actually this looks a bit sketchy! Why do we have to `Pin::new` this?

Reply:
The poll method on a `JoinHandle` accepts `Pin<&mut Self>`, but `self.0` is not pinned even though `self` is pinned, so i pinned it. 

I did more reading, and I found `pin_project` which allows you to _project_ the pin on a struct into it's fields, and don't need to `Pin::new`. Technically it was fine to `Pin::new` it since it is `Unpin`? Need to read up more on pin and unpin though.